src/ir/possible-contents.cpp

```diff
@@ -0,0 +1,1607 @@
+      auto tag = curr->catchTags[tagIndex];
+      auto* body = curr->catchBodies[tagIndex];
+
+      auto params = getModule()->getTag(tag)->sig.params;
+      if (params.size() == 0) {
+        return;
+      }
+
+      // Find the pop of the tag's contents. The body must start with such a
+      // pop, which might be of a tuple.
+      FindAll<Pop> pops(body);
+      assert(!pops.list.empty());
+      auto* pop = pops.list[0];
+      assert(pop->type.size() == params.size());
+      for (Index i = 0; i < params.size(); i++) {
+        if (isRelevant(params[i])) {
+          info.links.push_back(
+            {TagLocation{tag, i}, ExpressionLocation{pop, i}});
+        }
+      }
+
+#ifndef NDEBUG
+      // This pop was in the position we can handle, note that (see visitPop
+      // for details).
+      handledPops++;
+#endif
+    }
+  }
+  void visitThrow(Throw* curr) {
+    auto& operands = curr->operands;
+    if (!isRelevant(operands)) {
+      return;
+    }
+
+    auto tag = curr->tag;
+    for (Index i = 0; i < curr->operands.size(); i++) {
+      info.links.push_back(
+        {ExpressionLocation{operands[i], 0}, TagLocation{tag, i}});
+    }
+  }
+  void visitRethrow(Rethrow* curr) {}
+
+  void visitTupleMake(TupleMake* curr) {
+    if (isRelevant(curr->type)) {
+      for (Index i = 0; i < curr->operands.size(); i++) {
+        info.links.push_back({ExpressionLocation{curr->operands[i], 0},
+                              ExpressionLocation{curr, i}});
+      }
+    }
+  }
+  void visitTupleExtract(TupleExtract* curr) {
+    if (isRelevant(curr->type)) {
+      info.links.push_back({ExpressionLocation{curr->tuple, curr->index},
+                            ExpressionLocation{curr, 0}});
+    }
+  }
+
+  // Adds a result to the current function, such as from a return or the value
+  // that flows out.
+  void addResult(Expression* value) {
+    if (value && isRelevant(value->type)) {
+      for (Index i = 0; i < value->type.size(); i++) {
+        info.links.push_back(
+          {ExpressionLocation{value, i}, ResultLocation{getFunction(), i}});
+      }
+    }
+  }
+
+  void visitReturn(Return* curr) { addResult(curr->value); }
+
+  void visitFunction(Function* curr) {
+    // Vars have an initial value.
+    for (Index i = 0; i < curr->getNumLocals(); i++) {
+      if (curr->isVar(i)) {
+        Index j = 0;
+        for (auto t : curr->getLocalType(i)) {
+          if (t.isDefaultable()) {
+            info.links.push_back(
+              {getNullLocation(t), LocalLocation{curr, i, j}});
+          }
+          j++;
+        }
+      }
+    }
+
+    // Functions with a result can flow a value out from their body.
+    addResult(curr->body);
+
+    // See visitPop().
+    assert(handledPops == totalPops);
+  }
+
+  // Helpers
+
+  // Handles the value sent in a break instruction. Does not handle anything
+  // else like the condition etc.
+  void handleBreakValue(Expression* curr) {
```

Comment:
You're right, it should be here too. Fixing.

Note that `isRelevant` is mostly an optimization - just to avoid passing around values that we don't want to optimize. So the fuzzer won't catch missing places...